plugin-locationcomponent/src/main/java/com/mapbox/maps/plugin/locationcomponent/LocationPuckManager.kt

```diff
@@ -48,11 +48,17 @@ internal class LocationPuckManager(
   private val onLocationUpdated: ((Point) -> Unit) = {
     lastLocation = it
     if (settings.locationPuck is LocationPuck3D) {
-      val latitude =
-        if (delegateProvider.mapProjectionDelegate.getMapProjection() == MapProjection.Globe) {
-          delegateProvider.mapCameraManagerDelegate.cameraState.center.latitude()
-        } else it.latitude()
-      lastMercatorScale = mercatorScale(latitude)
+      delegateProvider.getStyle { style ->
+        val projectionProperty = style.getStyleProjectionProperty("name")
+        val isGlobeProjection =
+          projectionProperty.kind == StylePropertyValueKind.CONSTANT &&
+            (projectionProperty.value.contents as String).uppercase() == "GLOBE"
```

Comment:
maybe move to constants?

nit: nice to add as extension:
```
fun projectProperty.isGlobe() = this.value.contents as String).uppercase() == "GLOBE"
```
or smth similar

Reply:
not sure it's worth it as it's used in 1 place only but will do for better readability